lib/elixir/lib/access.ex

```diff
@@ -53,7 +53,11 @@ defmodule Access do
     :maps.find(key, map)
   end
 
-  def fetch(list, key) when is_list(list) and is_atom(key) do
+  def fetch(list, key) when is_list(list) do
+    unless is_atom(key) do
+      raise ArgumentError,
+        "the Access for keywords expect the key to be an atom, got: " <> inspect(key)
+    end
```

Comment:
I've made it at first place but decided to keep `list`-related logic grouped, but have no strong feelings one way or the other. :smile: